### PR TITLE
Added facings support to sprite trails

### LINKED ISSUE #10823: Directional smoke trail support
Would allow modders to implement smoke, dust and wave trails that actually match the actor's or projectile's facing.

Would also spare me from shipping custom variants of Smoke, Missile and Bullet in my MechCommander mod.

### PR DESCRIPTION
**Pull request by reaperrr**
`Bullet`, `Missile` and `SmokeTrailWhenDamaged`.

Closes #10823.

Test case can be found at https://github.com/reaperrr/OpenRA/tree/smoke-facings-test

### CODE REVIEW

File: OpenRA.Mods.Common/Effects/Bullet.cs [hunk 1]
```diff
@@ -47,6 +47,9 @@ public class BulletInfo : IProjectileInfo
 		[Desc("Trail animation.")]
 		public readonly string Trail = null;
 
+		[Desc("Loop this sequence of TrailImage while this projectile is moving.")]
```
Comment (obrakmann) [inline OpenRA.Mods.Common/Effects/Bullet.cs]: `TrailImage` -> `Trail`, I guess.

**Comment by abcdefg30** [comment]
Just a question: What about supporting multiple `TrailSequence`s as we do for the bullet image?

**Comment by reaperrr** [comment]
> Just a question: What about supporting multiple TrailSequences as we do for the bullet image?

Makes sense to do that now I guess, otherwise we'll need another upgrade rule if this is done in a later PR.

**Comment by reaperrr** [comment]
Updated.

**Comment by abcdefg30** [comment]
Looks good to me, too. :+1: / :white_check_mark:

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/4b43b1f74a632aa8753045b5bdc0a99c7b645577...2fcc6b9bf04a29e34ff3ae654c8d1dede37105f6)

**Pull request merged**